internal/unionstore/rbt/rbt.go

```diff
@@ -347,12 +349,16 @@ func (db *RBT) Set(key []byte, value []byte, ops ...kv.FlagsOp) error {
 	// the NeedConstraintCheckInPrewrite flag is temporary,
 	// every write to the node removes the flag unless it's explicitly set.
 	// This set must be in the latest stage so no special processing is needed.
-	var flags kv.KeyFlags
+	flags := x.GetKeyFlags()
+	if flags == 0 && x.vptr.IsNull() && x.isDeleted() {
+		db.count++
+		db.size += int(x.klen)
```

Comment:
When reusing the deleted node, doesn't need to clean the key/value inside the node?

Reply:
The value is already cleaned by `RevertToCheckpoint`, so `x.vptr.IsNull() == true` here.

https://github.com/tikv/client-go/blob/70049ae310bfcc02bbd8f3929ec4fd779d697e7f/internal/unionstore/arena/arena.go#L347-L357

And the key is unchanged, we just reuse it.